### ISSUE TITLE
convert retinaface model to caffe error

### ISSUE THREAD

**Issue by xiakj** [szad670401/advanced-mxnet2caffe#4]

**Comment by xiakj** [comment]
how to convert retinaface mxnet model to caffe. can you list the conversion details? thank u.

**Comment by gasgallo** [comment]
```bash
python json2prototxt.py --mx-json model-symbol.json --cf-prototxt caffe_target.prototxt
python mxnet2caffe.py --mx-model model --mx-epoch 0 --cf-prototxt caffe_target.prototxt --cf-model caffe_target.caffemodel
```

**Comment by xiakj** [comment]
> ```shell
> python json2prototxt.py --mx-json model-symbol.json --cf-prototxt caffe_target.prototxt
> python mxnet2caffe.py --mx-model model --mx-epoch 0 --cf-prototxt caffe_target.prototxt --cf-model caffe_target.caffemodel
> ```

got it. thanks

**Comment by xiakj** [comment]
> ```shell
> python json2prototxt.py --mx-json model-symbol.json --cf-prototxt caffe_target.prototxt
> python mxnet2caffe.py --mx-model model --mx-epoch 0 --cf-prototxt caffe_target.prototxt --cf-model caffe_target.caffemodel
> ```

and what's more, have you realize the detection_output_layer for retinaface in caffe? can you share your talented work?

**Comment by gasgallo** [comment]
@xiakejiang what do you mean? The post processing of retinaface outputs?

**Comment by xiakj** [comment]
> @xiakejiang what do you mean? The post processing of retinaface outputs?
yes. I converted the retinaface to caffe successfully, now I am writing the last layer in order to get the final detection result. I imitate the detection_output_layer of SSD in caffe. It is kind of difficult.

**Comment by gasgallo** [comment]
@xiakejiang there are many implementations online, I'm using something similar to the one in [this](https://github.com/cholihao/Retinaface-caffe) repo

**Comment by xiakj** [comment]
> @xiakejiang there are many implementations online, I'm using something similar to the one in [this](https://github.com/cholihao/Retinaface-caffe) repo

I alse noticed this repo, but it seems missing detect.h file. whether the detect.cpp can run correctly without the detect.h file?

**Comment by bleakie** [comment]
I meet this error。
txt_file.write('    shape: {{ dim: {} dim: {} dim: {} dim: {} }}\n'.format(info['shape'][0],
KeyError: 'shape'

**Comment by 791136190** [comment]
@bleakie you can try：if str(node_i['op']) == 'data' change to if str(node_i['op']) == 'null' and str(node_i['name']) == 'data'，in json2prototxt.py at line 40.
